File: check_jsx.js

const fs = require('fs');
const content = fs.readFileSync('src/features/messages/screens/ChatsScreen.jsx', 'utf8');

const tags = [];
const pattern = /<(/?View|/?Pressable|/?Text)/g;
let match;
const stack = [];

while ((match = pattern.exec(content)) !== null) {
  const tag = match[1];
  if (tag.startsWith('/')) {
    const top = stack.pop();
    if (top !== tag.substring(1)) {
      console.log(`Mismatch: ${top} vs ${tag}`);
    }
  } else {
    stack.push(tag);
  }
}

if (stack.length > 0) {
  console.log(`Remaining on stack: ${stack.join(', ')}`);
} else {
  console.log('Balanced!');
}

File: check_jsx.py

def check_balance(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        content = f.read()
    
    tags = []
    # Simplified tag seeker
    import re
    # Find <View, </View, <Pressable, </Pressable, <Text, </Text
    pattern = re.compile(r'<(/?View|/?Pressable|/?Text)')
    matches = pattern.findall(content)
    
    stack = []
    for tag in matches:
        if tag.startswith('/'):
            if not stack:
                print(f"Error: popped empty stack with {tag}")
            else:
                top = stack.pop()
                if top != tag[1:]:
                    print(f"Mismatch: {top} vs {tag}")
        else:
            stack.append(tag)
    
    if stack:
        print(f"Remaining on stack: {stack}")
    else:
        print("Balanced!")

check_balance('d:/Mohit_pryvo/Dating_App/src/features/messages/screens/ChatsScreen.jsx')
